### ISSUE TITLE
Line Chart Point Style Not Applying for First Point of Every Group in LineChart - Kibana Version 8.12.0

### ISSUE THREAD

**Issue by mageshsankar** [elastic/elastic-charts#2460]
**Summary**
In Kibana version 8.12.0, the pointStyleAccessor is not correctly applying the specified point styles to the first data point of each group in a line chart using @elastic/charts. This issue did not exist in Kibana version 7.16.3, where the point styles were applied correctly.

**Steps to Reproduce**

1. Set up a line chart using @elastic/charts in Kibana version 8.12.0.
2. Define a pointStyleAccessor to customize the appearance of data points.
3. Populate the chart with multiple series of data points, ensuring each series has at least two data points.
4. Observe that the pointStyleAccessor does not apply the specified styles to the first point of each series.
5. Expected Behavior
6. The pointStyleAccessor should consistently apply the defined styles to all data points, including the first point of each series, as it did in Kibana version 7.16.3.

Actual Behavior
The first data point of each series does not have the styles specified in the pointStyleAccessor applied. Subsequent points in the series correctly reflect the defined styles.

Code Example
Below is a simplified version of the code that demonstrates the issue:

```
  const dummyData = [
        { date: "2024-01-01", codeLines: 800, project: "Project A1", version: "v1.0" },
        { date: "2024-06-02", codeLines: 1100, project: "Project A", version: "v1.1" },
        { date: "2024-02-01", codeLines: 900, project: "Project A", version: "v1.0" },
        { date: "2024-07-02", codeLines: 950, project: "Project B", version: "v1.1" },
        { date: "2024-08-03", codeLines: 1000, project: "Project C", version: "v1.2" }
    ];< 
  return (
           <Chart renderer="canvas">
                        <Settings showLegend legendPosition={Position.Bottom} theme={lineChartTheme} />
                        <Tooltip customTooltip={toolTip} placement={Placement.Right} offset={40} />
                        <Axis
                            id={htmlID("left")}
                            position={Position.Left}
                            title="Code Lines"
                            showOverlappingLabels
                            tickFormat={formatNumber}
                        />
                        <Axis
                            id="bottom"
                            title="Data Capture Date"
                            position={Position.Bottom}
                            tickFormat={tickFormat}
                        />
                        <LineSeries
                            id={htmlID("lines")}
                            xScaleType={ScaleType.Ordinal}
                            yScaleType={ScaleType.Linear}
                            xAccessor="date"
                            yAccessors={["codeLines"]}
                            data={dummyData}
                            splitSeriesAccessors={["project"]}
                            pointStyleAccessor={(datum) => ({ radius: 2.75, strokeWidth: 6 })}
                            lineSeriesStyle={{ line: { strokeWidth: 4, opacity: 0.2 } }}
                            curve={CurveType.LINEAR}
                        />
                    </Chart>)
}

```

  
**Environment Details**

1. Kibana Version: 8.12.0
2. Elastic Charts Version: As included in Kibana 8.12.0
3. Browser: All supported browsers
4. OS: Linux

**Additional Information**

1. This issue was not present in Kibana version 7.16.3.
2. The same code worked correctly in version 7.16.3, with the pointStyleAccessor applying styles to all data points, including the first point of each series.

**Suggested Fix**
Investigate the changes between Kibana version 7.16.3 and 8.12.0 related to the @elastic/charts library, specifically focusing on the pointStyleAccessor implementation. Ensure that the pointStyleAccessor correctly applies styles to all data points, including the first point of each series. Consider adding a test case to validate this behavior in future releases.

Attachments
Screenshot of the line chart showing the issue in Kibana 8.12.0.
![image](https://github.com/elastic/elastic-charts/assets/55440472/c4afd7bd-a58a-4b3e-b7e7-fa47ba86cd9d)

**Comment by mageshsankar** [comment]
@nickofthyme  I can still reproduce this issue in latest version v65.2.1, here is the [code sandbox link](https://codesandbox.io/p/devbox/elastic-charts-nx7w99?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clxa520or00073j6g6fxr26jv%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clxa520or00023j6gw8g671kr%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clxa520or00043j6gra3u6h89%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clxa520or00063j6giuwb40gd%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B40%252C60%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clxa520or00023j6gw8g671kr%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxa520or00013j6ghy86j7hq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clxa520or00023j6gw8g671kr%2522%252C%2522activeTabId%2522%253A%2522clxa520or00013j6ghy86j7hq%2522%257D%252C%2522clxa520or00063j6giuwb40gd%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxa520or00053j6gb032ubxo%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clxa520or00063j6giuwb40gd%2522%252C%2522activeTabId%2522%253A%2522clxa520or00053j6gb032ubxo%2522%257D%252C%2522clxa520or00043j6gra3u6h89%2522%253A%257B%2522id%2522%253A%2522clxa520or00043j6gra3u6h89%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxa520or00033j6gzvfymp0l%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522install%2522%252C%2522id%2522%253A%2522clxa5dfz9006o3j6g7kdsow78%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clxa5dfz9006o3j6g7kdsow78%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A20.681818181818187%257D)

**Comment by nickofthyme** [comment]
Ok I see the issue now. I opened https://github.com/elastic/elastic-charts/pull/2464 to fix the issue.

As far as a workaround, that is not easy. You basically need to remove all isolated points as defined by this check...

https://github.com/elastic/elastic-charts/blob/501781bf8eda3adcdbc6eeb67646164a28561b89/packages/charts/src/chart_types/xy_chart/rendering/points.ts#L225-L242

Which you could do by filling the missing data so that all `x` values across all `split` series value a non-nil `y` value, such as `0`. This would then prevent application of the isolated point style overrides avoiding the problem. But this has unintended consequences such as connecting all the 

![Zight Recording 2024-06-11 at 06 11 05 PM](https://github.com/elastic/elastic-charts/assets/19007109/1237f54f-f49f-44aa-84dc-6a5438fe3de2)

That being said, I can backport the fix to as far as possible ~`kibana@8.15.x` if that would help. Let me know.

**Comment by mageshsankar** [comment]
@nickofthyme  Once you've completed the change, can we upgrade elastic-charts to the most recent version, which will resolve the issues while keeping the same Kibana version 8.12.0, or do we need to upgrade Kibana to a more recent version?

**Comment by mageshsankar** [comment]
@nickofthyme  i can still reproduce the with elastic-charts with version "65.2.0", can you check in the [code sandbox link ](https://codesandbox.io/p/devbox/elastic-charts-nx7w99?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clxa520or00073j6g6fxr26jv%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clxa520or00023j6gw8g671kr%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clxa520or00043j6gra3u6h89%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clxa520or00063j6giuwb40gd%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B40%252C60%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clxa520or00023j6gw8g671kr%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxa520or00013j6ghy86j7hq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clxa520or00023j6gw8g671kr%2522%252C%2522activeTabId%2522%253A%2522clxa520or00013j6ghy86j7hq%2522%257D%252C%2522clxa520or00063j6giuwb40gd%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxa520or00053j6gb032ubxo%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clxa520or00063j6giuwb40gd%2522%252C%2522activeTabId%2522%253A%2522clxa520or00053j6gb032ubxo%2522%257D%252C%2522clxa520or00043j6gra3u6h89%2522%253A%257B%2522id%2522%253A%2522clxa520or00043j6gra3u6h89%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxa520or00033j6gzvfymp0l%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522install%2522%252C%2522id%2522%253A%2522clxa5dfz9006o3j6g7kdsow78%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clxa5dfz9006o3j6g7kdsow78%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A20.681818181818187%257D)

**Comment by nickofthyme** [comment]
Hey @mageshsankar this was fixed in charts version [`v66.0.2`](https://github.com/elastic/elastic-charts/releases/tag/v66.0.2). Did you need this in a specific kibana version? As of now it will be added to the latest kibana version when https://github.com/elastic/kibana/pull/185906 is merged.

**Comment by mageshsankar** [comment]
Still, I can reproduce the issue.
![image](https://github.com/elastic/elastic-charts/assets/55440472/672f701f-df44-4883-8f53-054042db24b3)

**Comment by nickofthyme** [comment]
@mageshsankar yes because the isolated point styles are still being assigned to those points but as of `v66.0.2` you can override these styles using the `pointStyleAccessor`, but you would need to override all of the styles that you want to change.

I see how this can be tedious so maybe we add a way to opt out of this isolated point styling.

**Comment by nickofthyme** [comment]
@mageshsankar based on your comments I've add the `enabled` option to disable the isolated point styles from being applied. See #2472.

**Comment by mageshsankar** [comment]
@nickofthyme  Which version of Kibana has this fix?

**Comment by nickofthyme** [comment]
These changes are in `main` as of https://github.com/elastic/kibana/pull/186836, and will be released in kibana `8.15.0`.
